Lombiq.Tests.UI/Services/WebDriverFactory.cs

```diff
@@ -18,34 +25,60 @@ public static class WebDriverFactory
 {
     private static readonly ConcurrentDictionary<string, Lazy<bool>> _driverSetups = new();
 
-    public static ChromeDriver CreateChromeDriver(BrowserConfiguration configuration, TimeSpan pageLoadTimeout) =>
-        CreateDriver(
-            () =>
-            {
-                var options = new ChromeOptions().SetCommonOptions();
+    public static Task<ChromeDriver> CreateChromeDriverAsync(BrowserConfiguration configuration, TimeSpan pageLoadTimeout)
+    {
+        // Force headless mode if we are in Linux without a working graphical environment.
+        if (!configuration.Headless && RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+        {
+            configuration.Headless = string.IsNullOrWhiteSpace(Environment.GetEnvironmentVariable("DISPLAY"));
+        }
```

Comment:
This looks like something for all drivers.

Reply:
Good point, I have moved it to `AtataFactory.CreateDriverAsync` before `WebDriverFactory.CreateChromeDriverAsync` and friends are called.